internal/app/service/user_email_service.go

```diff
@@ -0,0 +1,126 @@
+package service
+
+import (
+	"context"
+
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/constants"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/helpers"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/pkg/specs"
+	"github.com/joshsoftware/profile_builder_backend_go/internal/repository"
+	"go.uber.org/zap"
+)
+
+// UserEmailService is the interface for the user email service
+type UserEmailService interface {
+	SendUserInvitation(ctx context.Context, userID int, profileID int) error
+	UpdateInvitation(ctx context.Context, userID int, profileID int) error
+}
+
+// inviteUser sends an email to the user with the invitation link
+func (userService *service) SendUserInvitation(ctx context.Context, userID int, profileID int) (err error) {
+	tx, _ := userService.ProfileRepo.BeginTransaction(ctx)
+	defer func() {
+		txErr := userService.ProfileRepo.HandleTransaction(ctx, tx, err)
+		if txErr != nil {
+			err = txErr
+			return
+		}
+	}()
+
+	profile, err := userService.ProfileRepo.GetProfile(ctx, profileID, tx)
+	if err != nil {
+		zap.S().Error("Error getting profile by profile id: ", err)
```

Comment:
Log error in detail.

error while getting profile. profile id :%d, err:%v

Reply:
done